idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,411 @@
+"""
+Centrifugal compressor stage model.  This model is based on:
+
+Modekurti et al., (2017). "Design, Dynamic Modeling, and Control of a Multistage CO2
+				Compresor System." International Journal of Greenhouse Gas Control. v62.
+Created: April 2020
+__Author__ = "Quang Minh Le "
+"""
+
+from __future__ import division
+# import Pyomo
+from pyomo.environ import (ConcreteModel, SolverFactory, Reals, NonNegativeReals,
+    Var, Param, Expression, Constraint, SolverFactory, value, exp, log, sqrt)
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+
+# Import IDAES cores
+from idaes.core import (declare_process_block_class,
+                        FlowsheetBlock)
+from idaes.generic_models.unit_models.pressure_changer import (PressureChangerData,
+																ThermodynamicAssumption)
+
+from idaes.generic_models.properties.helmholtz.helmholtz import (
+    HelmholtzThermoExpressions as ThermoExpr
+)
+import idaes.logger as idaeslog
+from idaes.core.util import from_json, to_json, StoreSpec
+
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+import idaes
+import math
+
+
+_log = idaeslog.getLogger(__name__)
+
+@declare_process_block_class("InletStage", doc="Inlet stage of compressor model")
+class InletStageData(PressureChangerData):
+    # Pressure changer with isentropic compressor options
+    CONFIG = PressureChangerData.CONFIG()
+    CONFIG.compressor = True
+    CONFIG.get("compressor")._default = True
+    CONFIG.get("compressor")._domain = In([True])
+    CONFIG.thermodynamic_assumption = ThermodynamicAssumption.isentropic
+    CONFIG.get("thermodynamic_assumption")._default = \
+        ThermodynamicAssumption.isentropic
+    CONFIG.get('thermodynamic_assumption')._domain = \
+        In([ThermodynamicAssumption.isentropic])
+
+    ################################################
+
+    def build(self):
+        super(InletStageData, self).build()
+        config = self.config # shorter config pointer
+
+        # The thermodynamic expression writer object, te, writes expressions
+        # including external function calls to calculate thermodynamic quantities
+        # from a set of state variables.
+        te = ThermoExpr(blk=self, parameters=config.property_package)
+
+        if SolverFactory('ipopt').available():
+          solver = SolverFactory('ipopt')
+          solver.options = {'tol':1e-6,
+                            'max_iter': 100,
+                            'linear_solver': 'ma27'}
+        else:
+          solver = None
+
+        #####################################################
+        # Declare variables for the model
+
+        # Some shorter refernces to property blocks
+        properties_in = self.control_volume.properties_in
+        properties_out = self.control_volume.properties_out
+
+        
+        self.kappaT = Var(self.flowsheet().config.time,
+                            initialize=5.0e-7)
```

Comment:
1. Check indentation.
2. You should make use of Pyomo `units` here to define the units of measurement on all `Vars` and `Params`.

Reply:
Further, these will depend on the property package.  I'm being a bit lazy, but you should be able to see something like the flow coefficient in the steam turbine outlet stage for an example of how to set the units in a generic way.

Reply:
I added to the Vars and Params in the revised version. Please check them.